vm/src/object/core.rs

```diff
@@ -110,7 +110,9 @@ struct PyInner<T> {
     typeid: TypeId,
     vtable: &'static PyObjVTable,
 
-    typ: PyRwLock<PyTypeRef>, // __class__ member
+    typ: PyTypeRef,
+    type_override: OnceCell<PyRwLock<PyTypeRef>>,
```

Comment:
I agree this is the only way we can practically accept for now, but growing `PyObject` body is not very pleasure.